owncloudComLibrary/src/main/java/com/owncloud/android/lib/common/utils/OCFileLoggingTree.kt

```diff
@@ -52,7 +52,17 @@ class OCFileLoggingTree(
                 if (!it.mkdirs())
                     Log.e(LOG_TAG, "couldn't create ${it.absoluteFile}")
             }
-            val fileNameTimeStamp = SimpleDateFormat(LOG_FILE_TIME_FORMAT, Locale.getDefault()).format(Date())
+
+            var fileNameTimeStamp = SimpleDateFormat(LOG_FILE_TIME_FORMAT, Locale.getDefault()).format(Date())
+
+            if (it.list().isNotEmpty()) {
+                val lastDayDateLogFile = getDayFromLogFileName(it.list().last())
+                val newDayDateLogFile = getDayFromLogFileName(fileNameTimeStamp)
+                if (lastDayDateLogFile == newDayDateLogFile) {
+                    fileNameTimeStamp = getDateFromLogFileName(it.list().last())
+                }
+            }
```

Comment:
We should write quality code as long as we can, free of warnings. In this case, Android Studio is already adding several warnings in this snippet. The method `list()` returns a nullable array, so let's cover that:
```suggestion
            it.list()?.let { logFiles ->
                if (logFiles.isNotEmpty()) {
                    val lastDayDateLogFile = getDayFromLogFileName(logFiles.last())
                    val newDayDateLogFile = getDayFromLogFileName(fileNameTimeStamp)
                    if (lastDayDateLogFile == newDayDateLogFile) {
                        fileNameTimeStamp = getDateFromLogFileName(logFiles.last())
                    }
                }
            }
```